docs/ops/movement/Pad_12.md

```diff
@@ -0,0 +1,261 @@
+Shape(3, 4)
+```
+for all of the *pad_mode* attribute options:
+* `pad_mode = "constant"`
+* `pad_mode = "edge"`
+* `pad_mode = "reflect"`
+* `pad_mode = "symmetric"`
+```
+OUTPUT =
+[[ 6, 7 ]]
+Shape(1, 2)
+```
+### Mixed pads example:
+```
+pads_begin = [2, -1]
+pads_end = [-1, 3]
+
+DATA =
+[[1,  2,  3,  4]
+ [5,  6,  7,  8]
+ [9, 10, 11, 12]]
+Shape(3, 4)
+```
+* `pad_mode = "constant"`:
+```
+OUTPUT =
+[[0, 0, 0, 0, 0, 0],
+ [0, 0, 0, 0, 0, 0],
+ [2, 3, 4, 0, 0, 0],
+ [6, 7, 8, 0, 0, 0]]
+Shape(4, 6)
+```
+* `pad_mode = "edge"`:
+```
+OUTPUT Shape(4, 6) =
+[[2, 3, 4, 4, 4, 4],
+ [2, 3, 4, 4, 4, 4],
+ [2, 3, 4, 4, 4, 4],
+ [6, 7, 8, 8, 8, 8]]
+Shape(4, 6)
+```
+* `pad_mode = "reflect"`:
+```
+OUTPUT =
+[[10, 11, 12, 11, 10, 9],
+ [6,   7,  8,  7,  6, 5],
+ [2,   3,  4,  3,  2, 1],
+ [6,   7,  8,  7,  6, 5]]
+Shape(4, 6)
+```
+* `pad_mode = "symmetric"`:
+```
+OUTPUT =
+[[6, 7, 8, 8, 7, 6],
+ [2, 3, 4, 4, 3, 2],
+ [2, 3, 4, 4, 3, 2],
+ [6, 7, 8, 8, 7, 6]]
+Shape(4, 6)
+```
+
+
+**Attributes**
+
+* *pad_mode*
+
+  * **Description**: *pad_mode* specifies the method used to generate new element values.
+  * **Range of values**: Name of the method in string format:
+    * `constant` - padded values are equal to the value of the *pad_value* input, if input not provided zero value is padded.
+    * `edge` - padded values are copied from the respective edge of the input `data` tensor.
+    * `reflect` - padded values are a reflection of the input `data` tensor; values on the edges are not duplicated. `pads_begin[D]` and `pads_end[D]` must be not greater than `data.shape[D] – 1` for any valid `D`.
+    * `symmetric` - padded values are symmetrically added from the input `data` tensor. This method is similar to the `reflect`, but values on edges are duplicated. Refer to the examples above for more details. `pads_begin[D]` and `pads_end[D]` must be not greater than `data.shape[D]` for any valid `D`.
+  * **Type**: `string`
+  * **Required**: *yes*
+
+**Inputs**
+
+* **1**: `data` tensor of arbitrary shape and type *T*. **Required.**
+
+* **2**: `pads_begin` 1D tensor of type *T_INT*. Number of elements matches the shape rank of *data* input. Specifies the number of padding elements to add at the beginning of each axis. Negative value means removal of the corresponding dimensions. **Required.**
+
+* **3**: `pads_end` 1D tensor of type *T_INT*. Number of elements matches the shape rank of *data* input. Specifies the number of padding elements to add at the ending of each axis. Negative value means removal of the corresponding dimensions. **Required.**
+
+* **4**: `pad_value` scalar tensor of type *T*. Used with the `pad_mode = "constant"` only. All new elements are populated with this value or with 0 if input not provided. Shouldn't be set for other `pad_mode` values. **Optional.**
+
+
+**Outputs**
+
+* **1**: Output padded tensor of type *T* with dimensions `pads_begin[D] + data.shape[D] + pads_end[D]` for each `D` from `0` to `len(data.shape) - 1`.
```

Comment:
I'm not sure but maybe it should smth like this:
```suggestion
* **1**: Output padded tensor of type *T* with dimensions `max(pads_begin[D] + data.shape[D] + pads_end[D], 0)` for each `D` from `0` to `len(data.shape) - 1`.
```

I think the case with `shape[5, 7], pads_begin[-3,1], pads_end[-3,1]` might be in dynamic scenario at least. What output shape should be here? Seems like `shape[0, 9]` not `shape[-1, 9]`. Could you please clarify this moment? Thanks!

Reply:
Thank you for this comment. Good point, I've updated the spec with `max` according to the suggestion to avoid negative dimensions.